pkg/middleware/validate_action_url.go

```diff
@@ -0,0 +1,60 @@
+package middleware
+
+import (
+	"fmt"
+	"net/http"
+	"path"
+
+	"github.com/grafana/grafana/pkg/cmd/grafana-cli/logger"
+	contextmodel "github.com/grafana/grafana/pkg/services/contexthandler/model"
+	"github.com/grafana/grafana/pkg/setting"
+	"github.com/grafana/grafana/pkg/util"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+func ValidateActionUrl(cfg *setting.Cfg) web.Handler {
+	return func(c *contextmodel.ReqContext) {
+		// ignore local render calls
+		if c.IsRenderCall {
+			return
+		}
+
+		// TODO: only process same origin requests
```

Comment:
same origin check seems to be missing + adding some tests for the whole middleware would be a good idea

Reply:
in progress, haven't pushed up the test file yet